Update fuzzing infra to build and use the shared framework (#108555)
dotnet/runtime: src/libraries/Fuzzing/README.md

### Prerequisites

> Build the runtime if haven't already:

Reviewer: ```suggestion
Build the runtime if you haven't already:
```

```cmd

> ./build.cmd clr+libs -rc release

Reviewer: Why this change?

Using checked+debug was intentional so that we run with all debug asserts enabled, including in corelib

Author: Updated it mainly because before `+packs+host` was required to be built, but not now, just tried to show now it can run with a normal build, though your point to use checked for runtime by default makes sense, I can change that

Reviewer: Not having to build `packs+host` will be nice to reduce build times, but I would keep it on checked+debug.

Reviewer: though I suppose it doesn't really matter as much anymore, the build works no matter which one you build locally.

```

and install the SharpFuzz command line tool:
```cmd
dotnet tool install --global SharpFuzz.CommandLine
```

> [!TIP]

> > The project uses a release runtime + debug libraries configuration by default.
> > If you want to use a different configuration, make sure to also adjust the paths in `run.bat`.

Reviewer: ```suggestion
> Automated fuzzing runs use a checked runtime + debug libraries configuration by default.
> You can use any configuration locally, but `checked` is recommended when testing changes in `System.Private.CoreLib`
```

### Fuzzing locally

> Build the `DotnetFuzzing` fuzzing project, the project is self contained so it will produce `DotnetFuzzing.exe` and all libraries needed for run into the output.

Reviewer: ```suggestion
Build the `DotnetFuzzing` fuzzing project. It is self-contained, so it will produce `DotnetFuzzing.exe` along with a copy of all required libraries.
```

```cmd
cd src/libraries/Fuzzing/DotnetFuzzing

dotnet build
```

> Now can `run run.bat` that will finds the path to the `DotnetFuzzing.exe` and run it with required arguments `\path\DotnetFuzzing.exe prepare-onefuzz deployment`. The `prepare-onefuzz` command will create separate directories for each fuzzing target, instrument the relevant assemblies, and generate a helper script for running them locally.

Reviewer: ```suggestion
Now you can run `run.bat`, which will create separate directories for each fuzzing target, instrument the relevant assemblies, and generate a helper script for running them locally.
```

### Prerequisites

Build the runtime if you haven't already:
```cmd
./build.cmd clr+libs -rc release
```

and install the SharpFuzz command line tool:
```cmd
dotnet tool install --global SharpFuzz.CommandLine
```

> [!TIP]
> The project uses a `Release` runtime + `Debug` libraries configuration by default.
> Automated fuzzing runs use a `Checked` runtime + `Debug` libraries configuration by default.
> You can use any configuration locally, but `Checked` is recommended when testing changes in `System.Private.CoreLib`.

### Fuzzing locally

Build the `DotnetFuzzing` fuzzing project. It is self-contained, so it will produce `DotnetFuzzing.exe` along with a copy of all required libraries.

```cmd
cd src/libraries/Fuzzing/DotnetFuzzing

dotnet build
```

> Now you can run `run.bat`, which will create separate directories for each fuzzing target, instrument the relevant assemblies, and generate a helper script for running them locally.

Author: I can update with my other PR
